x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -110,7 +110,30 @@ public boolean supports(AuthenticationToken token) {
 
     @Override
     public X509AuthenticationToken token(ThreadContext context) {
-        return token(context.getTransient(PKI_CERT_HEADER_NAME), principalPattern, logger);
+        Object pkiHeaderValue = context.getTransient(PKI_CERT_HEADER_NAME);
+        if (pkiHeaderValue == null) {
+            return null;
+        }
+        assert pkiHeaderValue instanceof X509Certificate[];
+        X509Certificate[] certificates = (X509Certificate[]) pkiHeaderValue;
+        if (certificates.length == 0) {
+            return null;
+        }
+        X509AuthenticationToken token = new X509AuthenticationToken(certificates);
+        // the following block of code maintains BWC:
+        // When constructing the token object we only return it if the Subject DN of the certificate can be parsed by at least one PKI
+        // realm. We then consider the parsed Subject DN as the "principal" even though it is generally incorrect because when several
```

Comment:
```suggestion
        // realm. We then consider the parsed Subject DN as the "principal" even though it is potentially incorrect because when several
```
Other phrases like _"not correct in the general case"_ or _"not strictly correct"_ would be fine too.
"Not generally correct" implies that it will be wrong more often than correct, which probably isn't the  case.